assets/js/atomic/blocks/product-elements/price/block.js

```diff
@@ -9,127 +9,89 @@ import {
 
-	const wrapperClassName = classnames( className, {
-		[ `${ parentClassName }__product-price` ]: parentClassName,
-	} );
-
-	if ( ! product.id ) {
-		return <ProductPrice align={ align } className={ wrapperClassName } />;
-	}
+	const colorProps = useColorProps( props );
+	const typographyProps = useTypographyProps( props );
 
-	const colorClass = getColorClassName( 'color', color );
-	const fontSizeClass = getFontSizeClass( fontSize );
-	const saleColorClass = getColorClassName( 'color', saleColor );
-	const saleFontSizeClass = getFontSizeClass( saleFontSize );
-
-	const classes = classnames( {
-		'has-text-color': color || customColor,
-		'has-font-size': fontSize || customFontSize,
-		[ colorClass ]: colorClass,
-		[ fontSizeClass ]: fontSizeClass,
-	} );
-
-	const saleClasses = classnames( {
-		'has-text-color': saleColor || customSaleColor,
-		'has-font-size': saleFontSize || customSaleFontSize,
-		[ saleColorClass ]: saleColorClass,
-		[ saleFontSizeClass ]: saleFontSizeClass,
-	} );
+	const wrapperClassName = classnames(
+		'wc-block-components-product-price',
+		className,
+		colorProps.className,
+		{
+			[ `${ parentClassName }__product-price` ]: parentClassName,
+		}
+	);
 
 	const style = {
-		color: customColor,
-		fontSize: customFontSize,
+		...typographyProps.style,
+		...colorProps.style,
 	};
 
-	const saleStyle = {
-		color: customSaleColor,
-		fontSize: customSaleFontSize,
-	};
+	if ( ! product.id ) {
+		return (
+			<ProductPrice align={ textAlign } className={ wrapperClassName } />
+		);
+	}
 
 	const prices = product.prices;
 	const currency = getCurrencyFromPriceResponse( prices );
 	const isOnSale = prices.price !== prices.regular_price;
-	const priceClassName = isOnSale
-		? classnames( {
-				[ `${ parentClassName }__product-price__value` ]: parentClassName,
-				[ saleClasses ]: isFeaturePluginBuild(),
-		  } )
-		: classnames( {
-				[ `${ parentClassName }__product-price__value` ]: parentClassName,
-				[ classes ]: isFeaturePluginBuild(),
-		  } );
-	const priceStyle = isOnSale ? saleStyle : style;
+	const priceClassName = classnames( {
+		[ `${ parentClassName }__product-price__value` ]: parentClassName,
+		[ `${ parentClassName }__product-price__value--on-sale` ]: isOnSale,
+	} );
 
 	return (
 		<ProductPrice
-			align={ align }
+			align={ textAlign }
 			className={ wrapperClassName }
+			priceStyle={ style }
+			regularPriceStyle={ style }
+			priceClassName={ priceClassName }
 			currency={ currency }
 			price={ prices.price }
-			priceClassName={ priceClassName }
-			priceStyle={ isFeaturePluginBuild() ? priceStyle : {} }
 			// Range price props
 			minPrice={ prices?.price_range?.min_amount }
 			maxPrice={ prices?.price_range?.max_amount }
 			// This is the regular or original price when the `price` value is a sale price.
 			regularPrice={ prices.regular_price }
-			regularPriceClassName={ classnames( {
-				[ `${ parentClassName }__product-price__regular` ]: parentClassName,
```

Comment:
I guess that it is better to keep this class. If a user creates a custom CSS that targets this class, the user will lose the customization. What do you think?